src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs

```diff
@@ -38,10 +38,16 @@ public static async Task<TreeNode[]> Expand(string connectionString, SecurityTok
         {
             using (SqlConnection conn = new SqlConnection(connectionString))
             {
-                conn.AccessToken = accessToken?.Token;
-                conn.Open();
+                if(accessToken != null)
+                {
+                    conn.AccessToken = accessToken.Token;
+                    await conn.OpenAsync();
```

Comment:
Shouldn't you still be trying to open even if there isn't an access token? Ideally this code should be generic enough that it can be used for other scenarios - such as ones where there's a user id/password.

Reply:
Ideally yes, but unfortunately opening the sql auth based connections leads to connection exception in this line.
https://github.com/microsoft/sqltoolsservice/blob/73c2a75fbaa903ea50f135f6f99fbf24e07a3523/src/Microsoft.SqlTools.SqlCore/ObjectExplorer/SmoModel/DatabaseTreeNode.cs#L23
 I haven't investigated smo to see why that's happening but I am making this fix to unblock people on mwc

Reply:
The number of SMO connection issues you're hitting is definitely concerning...

But regardless - so these failures would only be something that people hit in development? Since we won't have sql login connections once this goes live. Just want to make sure, although I guess this isn't really blocking. 

But please at least add a comment here explaining why it's only being opened when there's a token. (and ideally add context on why you're even opening it here in the first place too)